### ISSUE TITLE
[508] Use button for upload

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/va.gov-team#36877]
## Description

The upload button currently being used is a `<label>` wrapping a `<span>` set to a role of button. The input file type is hidden. Given that this is to restyle the default upload button, I don't think we need to include a `<label>` nor use a `<span>` as a button.

Question: Does a hidden file input need an associated label?

![Dev tools showing hidden file input with a span with role type button inside a label next to it](https://user-images.githubusercontent.com/136959/153895348-ae6dbd9d-1336-4d95-8e56-1116e0c24242.png)

This pattern can be found in:
- FileField (form system)
- FileInput (component-library) - used by the Claims Status Tool

Also, we shouldn't be using inline styling (`style={{ display: 'none' }}`) on the file input

## Tasks

- [ ] Test replacing label + span with a button will fulfill a11y requirements
- [ ] Use `vads-u-display--none` class on the input instead of inline styling
- [ ] Implement code as necessary
- [ ] Update tests

## Definition of done

- [ ] All tasks complete
- [ ] All tests passing
- [ ] Screenreader handles upload button appropriately

**Comment by humancompanion-usds** [comment]
@k80bowman and @bkjohnson - Looks to me like this may be fixed by using File Input (https://design.va.gov/storybook/?path=/docs/components-fileinput--default). Some questions:

* Why is File Input not documented in design.va.gov? Is that intentional or oversight? I think we may have missed in our audit looking for components that are in Storybook and nowhere else (neither design.va.gov nor Sketch)
* Do we have plans to make File input a web-component? Or is this Forms System work?
* Is this instance one we can just go fix with File input?

**Comment by Mottie** [comment]
Actually, within the form system, the [`FileField` component is used, and it's using the `<span>`](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/forms-system/src/js/fields/FileField.jsx#L509-L538) as the button, as shown in the screenshot

**Comment by bkjohnson** [comment]
> Why is File Input not documented in design.va.gov? Is that intentional or oversight?

As far as I know that's not intentional - it must have been an oversight (but I did find [documentation](https://design.va.gov/patterns/forms/file-upload) on a file upload pattern). We [do plan on creating a web component for it](https://vfs.atlassian.net/wiki/spaces/DST/pages/2040332297/React+components+to+web+components+-+Priorities), but it's a low priority at the moment.

As Robin points out, this is an example of the forms system using its own components rather than things provided by the design system. It's possible that we could swap out the forms system's `FileField` for our own `FileInput`, but there may be application unit or e2e tests which use selectors or other things specific to `FileField`.

For example, here's a [cypress test](https://github.com/department-of-veterans-affairs/vets-website/blob/467428915c4116cd1a1eb60490441ea33675ea7b/src/applications/appeals/10182/tests/10182-notice-of-disagreement.cypress.spec.js#L53-L54) which targets the `.schemaform-file-uploading` class which is used in the [`FileField` definition](https://github.com/department-of-veterans-affairs/vets-website/blob/467428915c4116cd1a1eb60490441ea33675ea7b/src/platform/forms-system/src/js/fields/FileField.jsx#L299-L300). Removing that component and replacing it with our own could cause tests to start failing, or give false positives.

**Comment by humancompanion-usds** [comment]
Thanks! I think I follow now. This could potentially be considered another forms related (adjacent?) web-component that the Forms System could consume from us, though they certainly don't need it in the immediate future. Still, we said we'd prioritize what we do for that team so that we're not blocking them. Let's discuss on Monday.

**Comment by caw310** [comment]
@Mottie is this still an issue?

**Comment by Mottie** [comment]
Platform `FileField` is now using a `label` that wraps a `va-button`, so I think we can close this ticket. Eventually, it should be updated to use a `va-file-input-multiple` web component, but there are still some problems with that web component that need to be addressed
